kratos/utilities/search_utilities.h

```diff
@@ -543,6 +637,251 @@ class SearchUtilities
+                noalias(coordinates) = it_point->Coordinates();
+                if constexpr (std::is_same<TPointIteratorType, ModelPart::NodeIterator>::value || std::is_same<TPointIteratorType, ModelPart::NodeConstantIterator>::value) {
+                    rAllPointsIds[counter] = it_point->Id();
+                } else {
+                    rAllPointsIds[counter] = initial_id + counter;
+                }
+                for (i_coord = 0; i_coord < 3; ++i_coord) {
+                    rAllPointsCoordinates[3 * counter + i_coord] = coordinates[i_coord];
+                }
+                ++counter;
+            }
+        }
+
+        // Return the ranks
+        return all_points_ranks;
+    }
+
+    /**
+     * @brief Synchronizes points between different processes.
+     * @details Synchonously.
+     * @param itPointBegin Iterator pointing to the beginning of the range of points.
+     * @param itPointEnd Iterator pointing to the end of the range of points.
+     * @param rSearchInfo The class containing the result of the search.
+     * @param rBoundingBox The bounding box considered.
+     * @param ThresholdBoundingBox The threshold for computing is inside bounding box considered.
+     * @param rDataCommunicator Object for data communication between processes.
+     * @param NumberOfPoints Local number of points to be synchronized.
+     * @param TotalNumberOfPoints Total number of points across all processes.
+     * @param IndexItIsJustCounter If the index considered it it just a counter.
+     * @return A vector containing the sizes of data for each process.
+     * @tparam TPointIteratorType The type of the point iterator.
+     * @tparam TBoundingBoxType The type of the bounding box.
+     * @return The ids of all points.
+     */
+    template<typename TPointIteratorType, typename TBoundingBoxType>
+    static std::vector<IndexType> SynchronizePointsWithBoundingBox(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        DistributedSearchInformation& rSearchInfo,
+        const TBoundingBoxType& rBoundingBox,
+        const double ThresholdBoundingBox,
+        const DataCommunicator& rDataCommunicator,
+        const int NumberOfPoints,
+        const int TotalNumberOfPoints,
+        const bool IndexItIsJustCounter = false
+        )
+    {
+        // Initialize and resize vectors
+        rSearchInfo.Reserve(TotalNumberOfPoints);
+        std::vector<double> all_points_coordinates(TotalNumberOfPoints * 3);
+        std::vector<IndexType> all_points_ids(TotalNumberOfPoints);
+
+        // Sync all points first
+        std::vector<int> all_points_ranks = SynchronizePointsWithRanks(itPointBegin, itPointEnd, all_points_coordinates, all_points_ids, rDataCommunicator, NumberOfPoints, TotalNumberOfPoints, IndexItIsJustCounter);
+
+        // Some definitions
+        IndexType i_coord = 0;
+        array_1d<double, 3> coordinates;
```

Comment:
Similar comment, but in this function you are only using it in the serial condition so even better.